src/main.rs

```diff
@@ -0,0 +1,859 @@
+
+    consumer
+        .subscribe(&[&input_topic])
+        .expect("Can't subscribe to specified topic");
+
+    // Create the `FutureProducer` to produce asynchronously.
+    let producer: FutureProducer = ClientConfig::new()
+        .set("bootstrap.servers", &brokers)
+        .set("message.timeout.ms", "5000")
+        .create()
+        .expect("Producer creation error");
+
+    // Create the outer pipeline on the message stream.
+    let stream_processor = consumer.stream().try_for_each(|borrowed_message| {
+        let decoder = AvroDecoder::new(sr_settings.clone());
+        let mut encoder = AvroEncoder::new(sr_settings.clone());
+        let producer = producer.clone();
+        let output_topic = output_topic.to_string();
+        async move {
+            // Process each message
+            record_borrowed_message_receipt(&borrowed_message).await;
+            // Borrowed messages can't outlive the consumer they are received from, so they need to
+            // be owned in order to be sent to a separate thread.
+            let owned_message = borrowed_message.detach();
+            record_owned_message_receipt(&owned_message).await;
+            tokio::spawn(async move {
+                // The body of this block will be executed on the main thread pool,
+                // but we perform `expensive_computation` on a separate thread pool
+                // for CPU-intensive tasks via `tokio::task::spawn_blocking`.
+                let mqa_event =
+                    tokio::task::spawn_blocking(|| handle_dataset_event(owned_message, decoder))
+                        .await
+                        .expect("failed to wait for handle dataset-event");
+
+                match mqa_event {
+                    Ok(Some(evt)) => {
+                        let fdk_id = evt.fdk_id.clone();
+                        let schema_strategy = SubjectNameStrategy::RecordNameStrategy(
+                            String::from("no.fdk.mqa.MQAEvent"),
+                        );
+
+                        match encoder.encode_struct(evt, &schema_strategy) {
+                            Ok(encoded_payload) => {
+                                let record: FutureRecord<String, Vec<u8>> =
+                                    FutureRecord::to(&output_topic).payload(&encoded_payload);
+                                let produce_future = producer.send(record, Duration::from_secs(0));
+                                match produce_future.await {
+                                    Ok(delivery) => info!(
+                                        "{} - Produce mqa event succeeded {:?}",
+                                        fdk_id.clone(),
+                                        delivery
+                                    ),
+                                    Err((e, _)) => error!(
+                                        "{} - Produce mqa event failed {:?}",
+                                        fdk_id.clone(),
```

Comment:
Kan se ut som `clone` er unødvendig. Mulig fordi det den alltid bare blir brukt i én av `match` casene.
```suggestion
                                        fdk_id,
                                        delivery
                                    ),
                                    Err((e, _)) => error!(
                                        "{} - Produce mqa event failed {:?}",
                                        fdk_id,
```